VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/GetHashCodeRefersToMutableMember/GetHashCodeRefersToMutableMemberAnalyzer.cs

```diff
@@ -0,0 +1,182 @@
+            context.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.NamedType);
+
+        private void AnalyzeSymbol(SymbolAnalysisContext context)
+        {
+            var namedType = (INamedTypeSymbol)context.Symbol;
+            var semanticModel = context.Compilation.GetSemanticModel(context.Symbol.Locations[0].SourceTree);
+
+            IMethodSymbol getHashCode;
+            GetHashCodeSymbol(namedType, out getHashCode);
+            if (getHashCode == null) { return; }
+
+            var getHashCodeLocation = getHashCode.Locations.FirstOrDefault();
+            var root = getHashCodeLocation?.SourceTree.GetRoot(context.CancellationToken);
+            if (root == null)
+            {
+                return;
+            }
+
+            var getHashCodeNode = (MethodDeclarationSyntax)root.FindNode(getHashCodeLocation.SourceSpan);
+            var nodes = getHashCodeNode.DescendantNodes(descendIntoChildren: target => true);
+
+            var identifierNameNodes = nodes.OfType<IdentifierNameSyntax>(SyntaxKind.IdentifierName);
+            foreach (var node in identifierNameNodes)
+            {
+                var symbol = semanticModel.GetSymbolInfo(node).Symbol;
+                if (symbol == null)
+                {
+                    continue;
+                }
+
+                if (symbol.Kind == SymbolKind.Field)
+                {
+                    var fieldIsMutableOrStatic = FieldIsMutableOrStatic((IFieldSymbol) symbol);
+                    if (fieldIsMutableOrStatic.Item1)
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(Rule, getHashCode.Locations[0],
+                            fieldIsMutableOrStatic.Item2));
+                    }
+                }
+                else if (symbol.Kind == SymbolKind.Property)
+                {
+                    var propertyIsMutable = PropertyIsMutable((IPropertySymbol) symbol, context.CancellationToken);
+                    if (propertyIsMutable.Item1)
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(Rule, getHashCode.Locations[0],
+                            propertyIsMutable.Item2));
+                    }
+                }
+            }
+        }
+
+        private void GetHashCodeSymbol(INamedTypeSymbol symbol, out IMethodSymbol getHashCodeSymbol)
+        {
+            getHashCodeSymbol = null;
+
+            foreach (var member in symbol.GetMembers())
+            {
+                if (!(member is IMethodSymbol))
+                {
+                    continue;
+                }
+
+                var method = (IMethodSymbol)member;
+                if (method.MetadataName == nameof(GetHashCode) && !method.Parameters.Any())
+                {
+                    getHashCodeSymbol = method;
+                }
+            }
+        }
+        
+        private Tuple<bool, string> FieldIsMutableOrStatic(IFieldSymbol field)
```

Comment:
It feels more natural to have `Tuple<string, bool>` since the key is the property/fieldname